packages/discovery-react-components/src/utils/document/transformEnrichment.ts

```diff
@@ -2,23 +2,37 @@ import invoiceOntology from './ontology/invoices';
 import purchaseOrderOntology from './ontology/purchase_orders';
 import flattenDeep from 'lodash/flattenDeep';
 import { getId } from './idUtils';
-import { ENRICHMENTS, getEnrichmentName } from 'components/CIDocument/utils/enrichmentUtils';
+import { getEnrichmentName } from 'components/CIDocument/utils/enrichmentUtils';
 import {
   EnrichedHtml,
   Contract,
+  Invoice,
+  PurchaseOrder,
   Attributes,
   Relations,
   Metadata
 } from 'components/CIDocument/types';
 import { Ontology } from './ontology/types';
 
+const ontologyMapping = {
+  invoice: invoiceOntology,
+  purchase_order: purchaseOrderOntology
+};
+
 const modelMapping = {
   contract: setMetadata,
   invoice: setAttributesAndRelations,
   purchase_order: setAttributesAndRelations
 };
 
-function setMetadata(enrichedHtml: Contract): void {
+function getUpdatedEnrichment(enrichment: EnrichedHtml, enrichmentName: string): EnrichedHtml {
+  if (enrichmentName in modelMapping) {
```

Comment:
I'd probably prefer something like this for this function (using `lodash.get`):

```
let mapping = get(modelMapping, [enrichmentName]);
return mapping ? mapping(enrichment, enrichmentName) : enrichment;
```

Reply:
I think the existing one is more clear readable.